File: src/types/installer.ts
export type RGEStatus = 'valid' | 'expired' | 'not_verified';

export type Installer = {
  id: string;
  companyName: string;
  siret: string;
  siren: string;
  address: string;
  city: string;
  postalCode: string;
  contactName: string;
  contactEmail: string;
  contactPhone: string;
  rgeNumber: string;
  rgeValidUntil: Date;
  rgeStatus: RGEStatus;
  qualificationTypes: string[];
  active: boolean;
  activeDossiersCount: number;
  totalDossiersCount: number;
  contractReference?: string;
  createdAt: Date;
};

export type DossierMilestone = {
  id: string;
  name: string;
  status: 'completed' | 'current' | 'pending';
  completedAt?: Date;
  description?: string;
};

export type ValidationFeedback = {
  id: string;
  type: 'rejection' | 'correction_request';
  reason: string;
  corrections: CorrectionItem[];
  createdAt: Date;
  createdBy: string;
};

export type CorrectionItem = {
  documentType: string;
  issue: string;
  required: boolean;
};

export type UploadStep = 'select_operation' | 'upload_documents' | 'review' | 'confirmation';

export type UploadedFile = {
  id: string;
  file: File;
  documentType: string;
  status: 'uploading' | 'completed' | 'error';
  progress: number;
  url?: string;
};

export type ProcessCategory = 'all' | 'residential' | 'tertiary' | 'industrial';

export type CEEProcess = {
  code: string;
  name: string;
  description: string;
  category: ProcessCategory;
  requiredDocuments: RequiredDocument[];
};

export type RequiredDocument = {
  type: string;
  name: string;
  requirement: 'required' | 'optional' | 'conditional';
  description: string;
  acceptedFormats: string[];
  maxSize: number;
  allowMultiple: boolean;
};

File: backend/app/schemas/installer.py
"""Installer schemas."""
from datetime import date, datetime
from typing import Optional
from uuid import UUID
from pydantic import BaseModel


class InstallerCreate(BaseModel):
    """Installer creation schema."""
    company_name: str
    siret: str
    siren: str
    address: str
    city: str
    postal_code: str
    contact_name: str
    contact_email: str
    contact_phone: Optional[str] = None
    rge_number: Optional[str] = None
    qualifications: list[str] = []


class InstallerUpdate(BaseModel):
    """Installer update schema."""
    company_name: Optional[str] = None
    address: Optional[str] = None
    city: Optional[str] = None
    postal_code: Optional[str] = None
    contact_name: Optional[str] = None
    contact_email: Optional[str] = None
    contact_phone: Optional[str] = None
    rge_number: Optional[str] = None
    active: Optional[bool] = None


class InstallerResponse(BaseModel):
    """Installer response schema."""
    id: UUID
    user_id: Optional[UUID] = None
    company_name: str
    siret: str
    siren: str
    address: str
    city: str
    postal_code: str
    contact_name: str
    contact_email: str
    contact_phone: Optional[str] = None
    rge_number: Optional[str] = None
    rge_valid_until: Optional[date] = None
    rge_status: str
    qualifications: list[str]
    contract_reference: Optional[str] = None
    active: bool
    created_at: datetime
    updated_at: datetime
    
    class Config:
        from_attributes = True

